hoomd/SnapshotSystemData.cc

```diff
@@ -39,6 +39,28 @@ void SnapshotSystemData<Real>::replicate(unsigned int nx, unsigned int ny, unsig
     pair_data.replicate(n, old_n);
     }
 
+template<class Real> void SnapshotSystemData<Real>::wrap()
+    {
+    for (unsigned int i = 0; i < particle_data.size; i++)
+        {
+        auto const pos = particle_data.pos[i];
+        auto const frac = global_box.makeFraction(particle_data.pos[i]);
+        auto modulus_positive = [](auto x){ return std::fmod(std::fmod(x, Real(1.0)) + Real(1.0), Real(1.0)); };
+        auto const wrapped = vec3<Real>(
+            modulus_positive(frac.x),
+            modulus_positive(frac.y),
+            modulus_positive(frac.z)
+            );
+        particle_data.pos[i] = global_box.makeCoordinates(wrapped);
+        int3 img;
+        img.x = static_cast<int>(frac.x);
+        img.y = static_cast<int>(frac.y);
+        img.z = static_cast<int>(frac.z);
```

Comment:
Use this function from `extern/cudacpu_vector_functions.h` to initialize the thing you want, rather than creating an object and then assigning it values.
```suggestion
        auto const img = make_int3(
            static_cast<int>(frac.x),
            static_cast<int>(frac.y),
            static_cast<int>(frac.z));
```